### DISCUSSION TITLE
fromArray can only retrieve Array[0] into sheet, but this array contains 3 elements

### DISCUSSION THREAD

**Discussion by Lydiachl** [PHPOffice/PhpSpreadsheet#2252] [Q&A] [answered]
I'm trying to use fromArray() to setting a range of cells from array, however, it can can only set Array[0] into the sheet, but I need set all elements of this array into sheet.

<?php
header("Content-type:text/html;charset=UTF-8");

require 'vendor/autoload.php';

use PhpOffice\PhpSpreadsheet\Spreadsheet;
use PhpOffice\PhpSpreadsheet\Writer\Xlsx;

$reader = \PhpOffice\PhpSpreadsheet\IOFactory::createReader('Xlsx');
$reader->setReadDataOnly(TRUE);

$spreadsheet = $reader->load("Customer Data.xlsx");
$worksheet = $spreadsheet->getActiveSheet();

$highestRow = $worksheet->getHighestRow();
$highestColumn = $worksheet->getHighestColumn();

$CustomerA = Array();

for ($row = 2; $row <= $highestRow; $row++){
    $customer = $worksheet -> getCellByColumnAndRow(1 , $row)->getValue();
    if($customer == 'A'){
    // Read a row of data into an array
    $rowData = $worksheet->rangeToArray('A'.$row . ':' . $highestColumn . $row, NULL, false, FALSE);
    if($CustomerA == null){
        $CustomerA = $rowData;
    }else{
        array_push($CustomerA,$rowData);
    }
    var_dump($CustomerA);
    echo "<br/>";
    }
}
$newSpreadsheet = new \PhpOffice\PhpSpreadsheet\Spreadsheet();
$newSpreadsheet->getActiveSheet()
    ->fromArray(
        $CustomerA,
        NULL,
        'C3'
);
    $writer = new Xlsx($newSpreadsheet);
    $writer->save('A Customer.xlsx');
?>
[Customer Data.xlsx](https://github.com/PHPOffice/PhpSpreadsheet/files/6937816/Customer.Data.xlsx)
[A Customer.xlsx](https://github.com/PHPOffice/PhpSpreadsheet/files/6937818/A.Customer.xlsx)

**Comment by oleibman** [comment, marked as answer]
First, you can make your code readable by placing `` ``` php `` before the code, and `` ``` `` after.
As for your problem, you can get the result you're looking for by replacing the following lines:
``` php
        if($CustomerA == null){
            $CustomerA = $rowData;
        } else {
            array_push($CustomerA,$rowData);
        }
```
with the following line:
``` php
        $CustomerA[] = $rowData[0];
```

**Reply by Lydiachl** [reply]
Thanks!!! It works. I'll make a better code format, not very familiar with it as it's the first time I tried it. Thanks again!
